htdocs/fourn/commande/list.php

```diff
@@ -132,6 +132,7 @@
 	$search_status = join(',', GETPOST('search_status', 'array:intcomma'));
 } else {
 	$search_status = (GETPOST('search_status', 'intcomma') != '' ? GETPOST('search_status', 'intcomma') : GETPOST('statut', 'intcomma'));
+	if ( is_array($search_status) ) $search_status = join(',', $search_status);
```

Comment:
If search_status is an array, we should go into line 132 after test in line 131.
Do you have param in url with param name "statut" instead of "search_status" ?

Reply:
@eldy  The problem occurs when you click the "back to list" link.  This link directs the end-user to the list with a url param named "restore_last_search

Bug Replication

1. Go to the Purchase orders list page.

2. Search for purchase orders with 2 or more status filters selected.

3. Click on one of the search results to view the card; The URL will contain a query parameter "save_lastsearch_values=1"

4. Click the "Back to list" link on the purchase order card; The URL will contain a query parameter "restore_lastsearch_values=1"

<img width="1437" alt="image" src="https://github.com/Dolibarr/dolibarr/assets/445543/1c9667f3-b059-4228-92d6-94a5af05833a">

Reply:
@eldy 
It appears that the "restore_lastsearch_values" query parameters restores the previous search values into the "$_POST" variable.  As a result, the "GETPOSTISARRAY" function call only checks "$_GET" where the parameter is not found.  Therefore, it falls to the else-block.

Reply:
@eldy  I have not identified the code-logic that processes "restore_lastsearch_values".  I just know that GETPOST finds the previous search value.

Reply:
Thanks. I pushed a better fix that should work on other pages too. Thanks to test it for your case.

Reply:
@eldy I tested the alternate fix in my sandbox and it works to resolve the issue reported.